paper.tex

```diff
@@ -206,11 +206,23 @@ \subsubsection{Participants}
 \subsubsection{Components}
 \begin{itemize}[leftmargin=*,labelsep=5.8mm]
 \item \textbf{Mask Authenticated Messaging: }
-IOTA\cite{IOTAwhitepaper} is a feeless and scalable cryptocurrency while MAM is the second layer data communication protocol built on top of Tangle.
+IOTA\cite{IOTAwhitepaper} is a feeless and scalable cryptocurrency while MAM is the second layer data communication protocol built on top of the IOTA network, the Tangle. MAM resolves the challenge of publishing authenticated streaming data as zero-value transactions to distributed ledgers, and allows access control of the data through encryption keys. 
 
-MAM resolves the challenge to publish encrypted streaming data to distributed ledgers. It publishes messages encrypted with a \textbf{session key} to the \textbf{channel}, which is a form of transaction where each address can be derived from the previous one. Therefore, with each \textbf{channel root}—the first message of a channel and the session key—all data on the channel is accessible.
+In IOTA protocol, seed is the identifier of its owner. An IOTA seed represents the ownership of all things associated with the user in the IOTA ecosystem, such as IOTA tokens or messages on the Tangle. With the seed, the owner can produce addresses and signatures in order to issue transactions or publish messages to MAM channels and endpoints.
 
-One can create multiple channels, however, the size of a channel is fixed which is decided before creation. Thus data providers need to firstly decide how to distribute data product into MAM channels. While the creation of a MAM channel is time-consuming, brokers are also responsible for channel creation, encrypted data publishing and session key certification in our system. See Figure \ref{fig:launching_product}
+Either channel or endpoint are like singly linked-list. The MAM channel's ID and endpoint's ID are the addresses of transaction and are the root of a channel and endpoint. Each address of a message can be derived from the previous one. A user can create multiple channels and multiple endpoints under the same channel. The messages can be encrypted with the session key and broadcasted in chronological order to the Tangle by attaching it to an endpoint. This allows only entities that know the session key can decode these messages after retrieving it from the Tangle, and also implements the concept of forward secrecy, where one have no access to the data back from his/hers entry point. Fig. shows the architecture of MAM.
```

Comment:
Either channel or endpoint "is"

Reply:
Should be "singly-linked list", not "singly linked-list"